nats/tests/sub_command_test.go

```diff
@@ -17,30 +17,76 @@ import (
 	"encoding/json"
 	"fmt"
 	"os"
+	"path/filepath"
 	"regexp"
 	"strings"
 	"testing"
+	"time"
+
+	"github.com/nats-io/nats-server/v2/server"
 
 	"github.com/nats-io/jsm.go"
 	"github.com/nats-io/nats.go"
 )
 
-func TestSubscribe(t *testing.T) {
+const (
+	primaryTestMsgData      = "primary test string"
+	secondaryTestMsgData    = "secondary test string"
+	nonJetstreamTestMsgData = "not jetstream test string"
+)
+
+func setupSubCommand(t *testing.T) (*server.Server, *nats.Conn, *jsm.Manager) {
+	t.Helper()
 	srv, nc, mgr := setupJStreamTest(t)
-	testMsgData := "this is a test string"
-	defer srv.Shutdown()
 
 	_, err := mgr.NewStream("ORDERS", jsm.Subjects("ORDERS.*"))
 	if err != nil {
 		t.Fatalf("unable to create stream: %s", err)
 	}
 	msg := nats.Msg{
 		Subject: "ORDERS.1",
-		Data:    []byte(testMsgData),
+		Data:    []byte(primaryTestMsgData),
+	}
+
+	err = nc.PublishMsg(&msg)
+	if err != nil {
+		t.Fatalf("unable to publish message: %s", err)
+	}
+
+	return srv, nc, mgr
+}
+
+func setupSubCommandNoJetstream(t *testing.T) (*server.Server, *nats.Conn) {
```

Comment:
feels like we need to make a few standard says to setup servers and use it in all tests.

Like, this one isnt really `sub` specific but its in the name

Reply:
Yeah, I agree. I've extracted another generic setup function into nats_test.go and moved some of the logic into the test itself. Let me know what you think. I can update the other tests in another PR if this works for you.